### DISCUSSION TITLE
Show folder/shortcut titles (Option)?

### DISCUSSION THREAD

**Discussion by hejchristiandev** [rxhanson/Charmstone-Community#1] [Ideas]
Love the idea of this app, there is just one thing I would like to see added that would really help me and possibly others too.

In its current state there is no distinguishing between normal folders (Which is what I hoped to use this tool for, as I could benefit from quicker access to some frequent folders I work in)

I understand it's a spatial memory app and so I'm supposed to remember which movement leads where. But I am struggling at the moment to build up that memory without a hint. I think the option to show folder titles (or any shortcut title even) could help here.

![CleanShot 2022-02-11 at 11 27 52](https://user-images.githubusercontent.com/29666588/153583875-d7d12116-7fb4-487e-99fe-270ec8436118.png)

**Comment by rxhanson** [comment]
Thanks for the feedback. 

Currently, when Charmstone is displayed it will show your color tags on folders, if you configured them in Finder. I realize that this is subtle and not communicated within the app, and I do have plans to improve this.

My initial thoughts were to add the first letter of the folder name on the folder icon, since there's not much room for a folder name and it avoids small text. Of course, you could still have an issue with common first letters, where I think color tags would be more beneficial. I was also thinking about tinting the folder with the color of the tag if there's only one color tag.

**Comment by ddobrinskiy** [comment]
Another option is to replace your folder icons from default to custom

<img width="261" alt="Screenshot 2023-03-04 at 22 02 47" src="https://user-images.githubusercontent.com/29500178/222919076-04d2510c-67a3-4c56-a065-b2d2cd7d34e1.png">

**Comment by rxhanson** [comment]
Thanks @ddobrinskiy. This discussion is slightly outdated since I had not mentioned that folders will now display an abbreviation of the folder name in the current version of Charmstone.

To replace folder icons, right click on the folder and select "Get Info", then drag and drop a new icon onto the icon there.

**Reply by K-blanc** [reply]
Just got to know your app. Looks great! But I'm having the same issue: just seeing folder icons, get the folder names would be a game changer. (For me a next step would be to get the option to populate dynamically, like get the top 3 recent folders).

**Reply by rxhanson** [reply]
Thanks for the feedback. The behavior is supposed to be that an abbreviation of the folder name for generic folders is  superimposed on the folder icon in a similar style to the other folders that have icons on them, but that was broken. I have a fix going in for the next release and am planning to get it out the door soon.
